# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/DOMIntersectionObserver.h
# Commit: 95679f6ce254
# Full Hash: 95679f6ce2544b21ec9cfd0d19961058d60155d6
# Author: Frederic Wang <fred.wang@free.fr>
# Date: 2023-11-27 17:00:06
# Regressor Bug: 1807253
# File Overlap Count: 1
# Description:
#   Bug 1807253 - unreliable timing for content-visibility:auto, r=emilio
#   
#   Rewrite implementation of content-visibility: auto as defined in
#   https://github.com/w3c/csswg-drafts/issues/8542
#   
# ==============================================================================

diff -r 9ea04a2cc829 -r 95679f6ce254 dom/base/DOMIntersectionObserver.h
--- a/dom/base/DOMIntersectionObserver.h	Mon Nov 27 08:00:41 2023 +0000
+++ b/dom/base/DOMIntersectionObserver.h	Mon Nov 27 08:26:46 2023 +0000
@@ -152,10 +152,10 @@
       const Document& aDocument, const nsINode* aRoot,
       const StyleRect<LengthPercentage>* aRootMargin);
 
-  enum class IsContentVisibilityObserver : bool { No, Yes };
+  enum class IsForProximityToViewport : bool { No, Yes };
   static IntersectionOutput Intersect(
       const IntersectionInput&, const Element&,
-      IsContentVisibilityObserver = IsContentVisibilityObserver::No);
+      IsForProximityToViewport = IsForProximityToViewport::No);
   // Intersects with a given rect, already relative to the root frame.
   static IntersectionOutput Intersect(const IntersectionInput&, const nsRect&);
 
@@ -165,9 +165,6 @@
   static already_AddRefed<DOMIntersectionObserver> CreateLazyLoadObserver(
       Document&);
 
-  static already_AddRefed<DOMIntersectionObserver>
-  CreateContentVisibilityObserver(Document&);
-
   static Maybe<nsRect> EdgeInclusiveIntersection(const nsRect& aRect,
                                                  const nsRect& aOtherRect);
 